plug.vim

```diff
@@ -1988,8 +1980,9 @@ function! s:shellesc_cmd(arg)
   return '^"'.substitute(escaped, '\(\\\+\)$', '\1\1', '').'^"'
 endfunction
 
-function! s:shellesc(arg)
-  if &shell =~# 'cmd.exe$'
+function! s:shellesc(arg, ...)
+  let shell = get(a:000, 0, s:is_win ? 'cmd.exe' : 'sh')
```

Comment:
Going forward, adding support for a new shell is a matter of adding a new `shellescape` function and a regex pattern to identify the shell based on the `&shell` path.